scripts/py_matter_idl/examples/matter_idl_plugin/__init__.py

```diff
@@ -0,0 +1,326 @@
+        case "int8s":
+            return "int32"
+        case "int16s":
+            return "int32"
+        case "int24s":
+            return "int32"
+        case "int32s":
+            return "int32"
+
+        case "int40s":
+            return "int64"
+        case "int48s":
+            return "int64"
+        case "int56s":
+            return "int64"
+        case "int64s":
+            return "int64"
+
+        case "long_char_string":
+            return "string"
+        case "char_string":
+            return "string"
+
+        case "long_octet_string":
+            return "bytes"
+        case "octet_string":
+            return "bytes"
+
+        case "single":
+            return "float"
+        case "float":
+            return "float"
+        case "double":
+            return "double"
+
+    return s
+
+
+# Python enum of types
+class DataType:
+    UINT = 1
+    INT = 2
+    BOOL = 3
+    CHAR_STRING = 4
+    OCT_STRING = 5
+    STRUCT = 6
+    FLOAT = 7
+    DOUBLE = 8
+
+
+def typeToTypeNum(s):
+    match(s):
+        case "uint32": return DataType.UINT
+        case "uint64": return DataType.UINT
+        case "int32": return DataType.INT
+        case "int64": return DataType.INT
+        case "bool": return DataType.BOOL
+        case "string": return DataType.CHAR_STRING
+        case "bytes": return DataType.OCT_STRING
+        case "float": return DataType.FLOAT
+        case "double": return DataType.DOUBLE
+
+    return DataType.STRUCT
+
+
+def commandArgs(command: Command, cluster: Cluster):
+    # Find command.input_arg in cluster.structs
+    for struct in cluster.structs:
+        if struct.name == command.input_param:
+            return struct.fields
+
+    return []
+
+
+def commandResponseArgs(command: Command, cluster: Cluster):
+    # Find command.input_arg in cluster.structs
+    for struct in cluster.structs:
+        if struct.name == command.output_param:
+            return struct.fields
+
+    return []
+
+
+def toTypeEncodedTag(tag, typeNum):
+    tag = (int(typeNum) << 19) | int(tag)
+    #tag = tag + 1
+    return tag
+
+
+def toFieldType(field: Field):
+    prefix = ""
+    protobufType = toProtobufType(field.data_type.name)
+    if field.is_list:
+        prefix = "repeated " + prefix
+    elif field.is_optional:
+        prefix = "optional " + prefix
+    return prefix + protobufType
+
+
+def toFieldTag(field: Field):
+    protobufType = toProtobufType(field.data_type.name)
+    typeNum = typeToTypeNum(protobufType)
+    tag = toTypeEncodedTag(field.code, typeNum)
+    return tag
+
+
+def toFieldComment(field: Field):
+    protobufType = toProtobufType(field.data_type.name)
+    typeNum = typeToTypeNum(protobufType)
+    tagComment = "/** %s Type: %d IsList: %d FieldId: %d */" % (
+        field.data_type.name, typeNum, field.is_list, field.code)
+    return tagComment
+
+
+class CustomGenerator(CodeGenerator):
+    """
+    Generation of cpp code for application implementation for matter.
```

Comment:
this does not seem to be about cpp code. Also __init__ mentions java which also seems wrong.

Can this be called something else than CustomGenerator? ProtobufGenerator?